lib/ios-log.js

```diff
@@ -0,0 +1,202 @@
+import path from 'path';
+import _ from 'lodash';
+import which from 'which';
+import glob from 'glob';
+import { fs, getLogger } from 'appium-logger';
+import _mkdirp from 'mkdirp';
+import touch from 'touch';
+import xcode from 'xcode'; 
+import B from 'bluebird';
+import { SubProcess } from 'teen_process';
+
+// Date-Utils: Polyfills for the Date object
+require('date-utils'); 
+
+let START_TIMEOUT = 10000;
+let logger = getLogger('IOS-log');
+let mkdirp = B.promisify(_mkdirp);
+
+class IOSLog { 
+  constructor(opts) {
+    this.udid = opts.udid;
+    this.simUdid = opts.simUdid;
+    this.showLogs = opts.showLogs;
+    this.proc = null;
+    this.iosLogStarted = false;
+    this.iosLogStartTime = null;
+    this.loggingModeOn = true;
+    this.logs = [];
+    this.logRow = '';
+    this.logsSinceLastRequest = [];
+  }
+
+  async startCaptureRealDevice () {
+    this.loggingModeOn = false;
+    let spawnEnv = _.clone(process.env);
+    logger.debug('Attempting iOS device log capture via libimobiledevice idevicesyslog');
+    let err = await which('idevicesyslog');
+    if (!err) { 
+      this.proc = new SubProcess('idevicesyslog', {env: spawnEnv});
+    } else {
+      logger.warn('Could not capture device log using libimobiledevice idevicesyslog.' +
+          ' Libimobiledevice probably is not installed');
+      logger.debug('Attempting iOS device log capture via deviceconsole');
+      let limDir = path.resolve(__dirname, '../../../build/deviceconsole');
+      spawnEnv.PATH = `${process.env.PATH}:${limDir}`;
+      spawnEnv.DYLD_LIBRARY_PATH = `${limDir}:${process.env.DYLD_LIBRARY_PATH}`;
+      this.proc = new SubProcess('deviceconsole', ['-u', this.udid], {env: spawnEnv});
+    }
+    await this.finishStartingLogCapture();
+  }
+
+  async startCapture () {
+    if (this.udid) { // if we have a real device
+      await this.startCaptureRealDevice();
```

Comment:
Do you want to do the rest of this process when it is a real device? If not, you need to `return this.startCaptureRealDevice();`.